Automattic/sensei
# Get lesson featured video thumbnail

### Changes proposed in this Pull Request

- This PR adds a hook on `save_post` for lessons that will store the video thumbnail URL in post meta.
- It supports:
   - YouTube
   - Vimeo
   - VideoPress
   - The WP Media Library
- It also supports the Classic Editor for cases where people are still using the `_lesson_video_embed` meta.
- I also added a method named `get_featured_video_thumbnail_url` to easily get the postmeta value to use in the Video templates later.

### Testing instructions

* Check this PR out
* Add a Lesson
* Add a Featured Video Block
* Add a YouTube Video like `https://www.youtube.com/watch?v=dQw4w9WgXcQ`
* Publish/Update the post
* Check the postmeta for that post to make sure `_featured_video_thumbnail` has the right URL there.
* Repeat the process for VideoPress, WP Media Library Videos, and Vimeo.

 ### Note about VideoPress

There are actually two ways that you can add VideoPress video's to your site - one is through embedding a VideoPress URL which creates an `core/embed` block.  The other is if you have a Jetpack site with VideoPress enabled and you upload a video directly to your site which is then converted to a VideoPress video but aliased to your local Media file - in that case it's a `core/video` block.  Feel free to test for both - if you want to test the Jetpack version you will need to use a site connected to Jetpack with VideoPress enabled.

### Screenshot / Video
![featured-video-thumb](https://user-images.githubusercontent.com/3220162/191377422-61026a2a-8af9-455c-bc8b-f8d28db61d62.gif)
